samples/msal-node-samples/standalone-samples/auth-code-pkce/index.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+const express = require("express");
+const msal = require('@azure/msal-node');
+
+const SERVER_PORT = process.env.PORT || 3000;
+
+// Before running the sample, you will need to replace the values in the config, 
+// including the clientSecret
+const config = {
+    auth: {
+        clientId: "89e61572-2f96-47ba-b571-9d8c8f96b69d",
+        authority: "https://login.microsoftonline.com/5d97b14d-c396-4aee-b524-c86d33e9b660"
+    },
+    system: {
+        loggerOptions: {
+            loggerCallback(loglevel, message, containsPii) {
+                console.log(message);
+            },
+            piiLoggingEnabled: false,
+            logLevel: msal.LogLevel.Verbose,
+        }
+    }
+};
+
+/**
+ * PKCE Setup
+ * 
+ * MSAL enables PKCE in the Authorization Code Grant Flow by including the codeChallenge and codeChallengeMethod parameters
+ * in the request passed into getAuthCodeUrl() API, as well as the codeVerifier parameter in the
+ * second leg (acquireTokenByCode() API).
+ * 
+ * Generating the codeVerifier and the codeChallenge is the client application's responsiblity.
+ * For this sample, you can either implement your own PKCE code generation logic or use an existing tool
+ * to manually generate a Code Verifier and Code Challenge, plugging them into the pkceCodes object below.
+ * 
+ * For details on implementing your own PKCE code generation logic, consult the 
+ * PKCE specification https://tools.ietf.org/html/rfc7636#section-4
+ */
+
+const PKCE_CODES = {
+    CHALLENGE_METHOD: "S256", // Use SHA256 Algorithm
+    VERIFIER: "", // Generate a code verifier for the Auth Code Request first
```

Comment:
I think it would be good to include a sample showing how to generate these values. You can use the node crypto libraries to do this.

Reply:
Node `crypto` APIs need some more work to be made public @pkanher617. We found some bugs and concluded the activity may need some more work.

We want the users to default to PKCE for public clients and hence this sample is being merged now, but we are tracking the activity to provide helper APIs with node crypto libraries later this quarter.

Reply:
Yeah that's not what I meant by this. You can easily generate PKCE parameters using the node crypto library in the sample. We can do it in a follow up but this sample is incomplete.